pytripgui/main_window_qt_vc/main_window_cont.py

```diff
@@ -0,0 +1,150 @@
+
+    def on_add_new_patient(self):
+        new_patient = PatientGui(self.model.kernels)
+        self.model.patients.append(new_patient)
+        return new_patient
+
+    def on_open_voxelplan(self):
+        path = self.view.browse_file_path("Open Voxelpan", "Voxelplan (*.hed)")
+        filename, extension = os.path.splitext(path)
+
+        if filename == "":
+            return
+
+        if self.model.current_patient:
+            patient = self.model.current_patient
+        else:
+            patient = self.on_add_new_patient()
+            self.model.patient_tree_cont.synchronize()
+
+        patient.open_ctx(filename + ".ctx")  # Todo catch exceptions
+        patient.open_vdx(filename + ".vdx")  # Todo catch exceptions
+
+        self.model.patient_tree_cont.synchronize()
+        self.model.one_plot_cont.set_patient(self.model.current_patient)
+
+    def on_add_new_plan(self):
+        if not self.model.current_patient:
+            message = InfoMessages["addNewPatient"]
+            self.view.show_info(message[0], message[1])
+            return
+
+        if not self.model.current_patient.ctx or not self.model.current_patient.vdx:
+            message = InfoMessages["loadCtxVdx"]
+            self.view.show_info(message[0], message[1])
+            return
+
+        if not self.model.kernels:
+            message = InfoMessages["configureKernelList"]
+            self.view.show_info(message[0], message[1])
+            return
+
+        self.model.current_patient.add_new_plan()
+        self.model.patient_tree_cont.synchronize()
+
+    def on_kernels_configuraotr(self):
+        """
+        Kernel dialog opened from window->settings->kernel
+        """
+        from pytripgui.kernel_vc import KernelController
+
+        model = self.model.kernels
+        view = self.view.get_kernel_config_view()
+
+        controller = KernelController(model, view)
+        controller.set_view_from_model()
+        view.show()
+
+        if controller.user_clicked_save:
+            self.settings.save()
+
+    def on_trip98_config(self):
+        """
+        Config menu opened from window->Settings->TRiP98 Config
+        """
+        logger.debug("TRiP config menu()")
+
+        from pytripgui.config_vc import ConfigController
+
+        view = self.view.get_trip_config_view()
+
+        controller = ConfigController(self.model.executor.trip_config, view)
+        controller.set_view_from_model()
+        view.show()
+
+        if controller.user_clicked_save:
+            self.settings.save()
+
+    def on_execute_plan(self, patient, plan):
+        if not plan.fields:
+            message = InfoMessages["addOneField"]
+            self.view.show_info(message[0], message[1])
+            return
+
+        if self.model.executor.check_config() != 0:
+            message = InfoMessages["configureTrip"]
+            self.view.show_info(message[0], message[1])
+            return
+
+        if plan.kernel.sis_path == "":
```

Comment:
better would be `if not plan.kernel.sis_path` as it would check for None value

Reply:
ok